### ISSUE TITLE
rosdep does not enforce version_gte tag

### ISSUE THREAD

**Issue by awesomebytes** [ros-infrastructure/rosdep#325]
Following this documentation:
http://www.ros.org/reps/rep-0127.html#build-depend-multiple

I added to my package.xml the line:

```
  <build_depend version_gte="1.10.6">sensor_msgs</build_depend>
```

And I executed (at my workspace, just over src folder):

```
rosdep install --from-paths src --ignore-src --rosdistro hydro
```

And I got:

```
#All required rosdeps installed succesfully
```

But my system ros-hydro-sensor-msgs is:

```
$ dpkg -l | grep sensor-msgs
ii  ros-hydro-sensor-msgs                        1.10.2-0precise-20131010-0201-+0000
```

The package itself is: https://github.com/pal-robotics/object_recognition_clusters just in case.

**Comment by mdhorn** [comment]
+1
I just ran into the same "problem".  I was using the newer, [format 2 of package.xml](http://www.ros.org/reps/rep-0140.html#build-depend-multiple), but I had the same expectation of rosdep.
I was surprised when I set version_gte in the package.xml and rosdep installed a lower version.

**Comment by peci1** [comment]
I agree that with Debian packages, there are just two options - update to latest, or fail (or ignore the version). But with pip, things get more interesting, because you can easily force version number.

I know that REP-140 has been there for a long time, but wouldn't it deserve a mention about rosdep not supporting these tags at all? Or at least a link to this issue? I know that other tools than rosdep may process the package manifest, but rosdep is the first I think about when I see a package manifest.

**Comment by realtime-neil** [comment]
What's the status of this issue? I wound up here after my (failed) attempts to get `rosdep` to pass version information to `apt`.

**Comment by FelixBlix** [comment]
I would also like to know the status of this issue?
Also; what is this "new tool" that is being refferenced? is it for ROS2 or is it something that will resolve this issue for ROS users as well?

**Comment by NikolausDemmel** [comment]
The status is that noone is working on this. Also for ROS2, there is currently no work towards replacing rosdep AFAIK.

A while ago there was work on a "new tool", xylem, which was intended as a replacement for rosdep with a ROS-independent core. I belive @wjwwood started it, then I worked on this for a while and together with @wjwwood and @tfoote we had some discussion on how this issue should be addressed in xylem (see https://xylem.readthedocs.io/en/latest/design.html#versions-in-rules-files). However, due to bandwidth limitations (I started a new job), this never reached beyond prototype stage and its not ready to replace rosdep. In particular version checking wasn't even implemented in the prototype, IIRC. From OSRF side, I guess rosdep has been working well enough, so other projects were more pressing.

**Comment by wjwwood** [comment]
That's an accurate summary.

**Comment by FelixBlix** [comment]
Thanks for the accurate and quick reply. We'll go back to using our own setup script then.
